fboom_me.sh

```diff
@@ -0,0 +1,201 @@
+# Plowshare fboom.me module
+# by idleloop <idleloop@yahoo.com>, v1.0, Feb 2016
+#
+# This file is part of Plowshare.
+#
+# Plowshare is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Plowshare is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Plowshare.  If not, see <http://www.gnu.org/licenses/>.
+#
+# (Official API probably similar to: https://github.com/keep2share/api)
+
+MODULE_FBOOM_ME_REGEXP_URL='https\?://\(www\.\)\?fboom.me/'
+
+MODULE_FBOOM_ME_DOWNLOAD_OPTIONS=""
+MODULE_FBOOM_ME_DOWNLOAD_RESUME=yes
+MODULE_FBOOM_ME_DOWNLOAD_FINAL_LINK_NEEDS_COOKIE=no
+MODULE_FBOOM_ME_DOWNLOAD_FINAL_LINK_NEEDS_EXTRA=
+MODULE_FBOOM_ME_DOWNLOAD_SUCCESSIVE_INTERVAL=
+
+MODULE_FBOOM_ME_PROBE_OPTIONS=""
+
+# Output an fboom_me file download URL
+# $1: cookie file
+# $2: fboom_me url
+# stdout: real file download link
+fboom_me_download() {
+    local -r COOKIE_FILE=$1
+    local URL BASE_URL FILE_NAME PRE_URL
+    local PAGE FORM_HTML FILE_ID WAIT
+
+    # get canonical URL and BASE_URL for this file
+    URL=$(curl -I "$2" | grep_http_header_location_quiet) || return
+    [ -n "$URL" ] || URL=$2
+    BASE_URL=${URL%/file*}
+    readonly URL BASE_URL
+
+    PAGE=$(curl -c "$COOKIE_FILE" "$URL") || return
```

Comment:
If you have an opportunity to use english language, then use it. This service has this opportunity. So before download function there should be excaly this static function (I have already checked how they change language):

```
# Static function. Switch language to english
# $1: cookie file
# $2: base URL
fboom_me_switch_lang() {
    curl "$2/site/setLanguage" -b "$1" -c "$1" -d 'language=en' > /dev/null || return
}

```

and before we load first page this code:
```
fboom_me_switch_lang "$COOKIE_FILE" "$BASE_URL"

PAGE=$(curl -b "$COOKIE_FILE" -c "$COOKIE_FILE" "$URL") || return
```

Important note: Here we use '-b "$COOKIE_FILE"', otherwise language settings will be removed.